src/main/java/nl/andrewl/emaildatasetbrowser/control/DatasetOpenAction.java

```diff
@@ -14,6 +14,7 @@
 
 public class DatasetOpenAction extends AbstractAction {
 	private static final String PREF_OPEN_DIR = "dataset_open_dir";
+	private static final String PREF_LAST_DS = "dataset_last_dataset_path";
```

Comment:
The purpose of this action is only to open a dataset, and I don't think it should be responsible for multiple different scenarios. If you want to centralize the logic for opening and closing a dataset somewhere else, that's fine, but the point is that an action should not expose public methods besides its constructor and `actionPerformed`.

On a separate note, I am not particularly in favor of trying to open the last dataset immediately when starting the app, as this can, depending on the size of the dataset, incur a significant startup delay of up to minutes, for datasets with millions or billions of emails.

Reply:
You make a fair point. I tend to not use the command line, so I don't use that functionality (I just double-click the ``.jar`` to open the program). 

I could expand a tiny bit on this by making auto-loading optional, have that set in a general settings panel or something, and then turn it off by default to avoid issues like you mentioned. I just use the same dataset over and over again, so it makes sense for me to want to auto-open the last dataset. 
When I do this, I'll separate the auto-open logic as well, as you suggested.